OvmfPkg/Library/PlatformInitLib/IntelTdx.c

```diff
@@ -635,8 +647,14 @@ ValidateHobList (
 
         break;
 
-      // EFI_HOB_GUID_TYPE is variable length data, so skip check
+      // EFI_HOB_GUID_TYPE is variable length data, the size of the TD Hob list had verify on the begain of the loop,

```

Comment:
EFI_HOB_TYPE_GUID_EXTENSION is variable-length Hob. The total size of the TdHob list is checked at the beginning of the loop. So we only need to check the min size of the HOB.